src/typ.rs

```diff
@@ -865,27 +928,31 @@ impl<'a> Env<'a> {
                 name,
                 args,
             } => {
-                let args = args
-                    .iter()
-                    .map(|t| self.type_from_ast(t, vars, new))
-                    .collect::<Result<Vec<_>, _>>()?;
-                let info = self
-                    .get_type_constructor(module, name)
-                    .map_err(|e| convert_get_type_constructor_error(e, &meta))?;
-                if args.len() != info.arity {
-                    return Err(Error::IncorrectTypeArity {
-                        meta: meta.clone(),
+                if let Ok(typ) = self.get_real_type(module, name) {
+                    return Ok(typ.to_owned());
+                } else {
```

Comment:
As we're returning in the `if branch we can remove the `else` branch and have the code as it was before.

Reply:
sure